### PR TITLE
fix(dashboard): kill elastic overscroll, reuse ChatPanel's bounce-prevention

### PR DESCRIPTION
**Pull request by jamesbrink**
## Problem

The Dashboard's scroll surface rubber-banded at its top and bottom edges on macOS WKWebView — the same elastic-overscroll behavior that ChatPanel already neutralized for the chat surface long ago. `ChatPanel.module.css` `.messages` pairs `overscroll-behavior-y: none` with the `usePreventScrollBounce` hook; `Dashboard.module.css` `.scrollBody` had neither.

## Why a naive copy doesn't work here

`usePreventScrollBounce` binds capture-phase wheel/touch listeners inside a `useEffect` that reads `containerRef.current` **once** on mount. ChatPanel gets away with this because `.messages` is a single stable element that's always mounted while ChatPanel is on screen.

Dashboard, by contrast, renders one of **three** mutually-exclusive branches at a time (project-scoped, no-workspaces, active-workspaces) and each branch has its own `<div className={styles.scrollBody}>`. Attaching the same `useRef` to all three branches and calling the hook once at the top of the component would only bind against whichever branch happened to mount first — every subsequent branch swap would silently leak the original listener and leave the new pane unprotected.

## Fix

Extract a tiny shared component, `BoundedScrollPane`, that owns its own ref and runs `usePreventScrollBounce` per-instance:

```tsx
export const BoundedScrollPane = forwardRef<HTMLDivElement, HTMLAttributes<HTMLDivElement>>(
  function BoundedScrollPane({ children, ...rest }, forwardedRef) {
    const innerRef = useRef<HTMLDivElement>(null);
    usePreventScrollBounce(innerRef);
    useImperativeHandle(forwardedRef, () => innerRef.current as HTMLDivElement);
    return <div ref={innerRef} {...rest}>{children}</div>;
  },
);
```

`forwardRef` + `useImperativeHandle` keep the escape hatch for consumers that need direct DOM access (`useStickyScroll`, search highlight scopes, future scroll-into-view helpers).

Each of Dashboard's three branches now renders `<BoundedScrollPane className={styles.scrollBody}>` instead of a bare `<div>`. When the user navigates between branches, React unmounts one instance (running the hook's cleanup) and mounts a fresh one (binding the listeners against its own node).

CSS gets the same belt-and-suspenders treatment as `.messages` — `overscroll-behavior-y: none` handles the trackpad-flick edge cases before the wheel listener intercepts.

## Tests

`src/ui/src/components/shared/BoundedScrollPane.test.tsx` (new — 4 cases):

- **renders a div with forwarded className + arbitrary HTML attrs** — guards the prop pass-through that every consumer relies on for styling and ARIA.
- **forwards the ref to the underlying DOM node** — pins the `useImperativeHandle` contract so consumers that need direct DOM access still get it.
- **calls preventDefault on a wheel at the scroll boundary** — end-to-end check that the hook is wired into the wrapper; a future refactor that decouples them would fail here loudly.
- **re-mounting via differing `key` re-binds against the new container** — this is the actual Dashboard motivating case: differing keys force React to unmount the first instance and mount a fresh one, and the test asserts the new pane's boundary is what controls cancellation. If the hook were still bound to the unmounted pane's closure, the wheel would pass through and the test would fail.

The bulk of the predicate coverage lives in the hook's own existing suite (`usePreventScrollBounce.test.tsx`), so this file pins only the wrapper-specific contract.

```
Test Files  158 passed (158)
Tests       1894 passed (1894)        (was 1890 on main; +4 new)
```

CI parity: `tsc -b`, `lint:css` (design-token + Monaco font-stack), and the existing ESLint pre-merge all clean.

## Follow-ups (out of scope)

Settings tabs, the MCP picker modal, and several list popovers have their own scroll surfaces that almost certainly have the same elastic behavior. They can adopt `BoundedScrollPane` directly — that's the reason this lives in `components/shared/` rather than as a private helper inside Dashboard.

### CODE REVIEW

File: src/ui/src/components/shared/BoundedScrollPane.tsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  HTMLDivElement,
+  HTMLAttributes<HTMLDivElement>
+>(function BoundedScrollPane({ children, ...rest }, forwardedRef) {
+  const innerRef = useRef<HTMLDivElement>(null);
+  usePreventScrollBounce(innerRef);
+  // `useImperativeHandle` lets callers pass a normal ref while we keep our
+  // own internal ref for the hook. Without this, consumers would have to
+  // choose between getting the DOM node and getting the bounce-prevention.
+  useImperativeHandle(forwardedRef, () => innerRef.current as HTMLDivElement);
+  return (
```
[the remark this answers is not in the document]
Comment (jamesbrink) [reply src/ui/src/components/shared/BoundedScrollPane.tsx]: Addressed in 517de882 — both concerns were valid. Rather than patch the symptoms (add a deps array, switch the cast to `!`), I dropped `useImperativeHandle` entirely in favor of the canonical merge-refs callback pattern:

```tsx
const setRefs = useCallback(
  (node: HTMLDivElement | null) => {
    innerRef.current = node;
    if (typeof forwardedRef === "function") {
      forwardedRef(node);
    } else if (forwardedRef) {
      (forwardedRef as MutableRefObject<HTMLDivElement | null>).current = node;
    }
  },
  [forwardedRef],
) satisfies Ref<HTMLDivElement>;
```

That sidesteps both issues:
- **No deps array question** — `useCallback` has explicit `[forwardedRef]`.
- **No cast** — React invokes a callback ref with the real `HTMLDivElement | null` on mount AND unmount, so the forwarded ref tracks the DOM node's actual lifecycle including the null cleanup transition. The previous `useImperativeHandle(... as HTMLDivElement)` would have lied to consumers about a value that legitimately can be null at lifecycle boundaries.

Added a regression test that pins the lifecycle contract — forwards to a `MutableRefObject`, mounts (`.current === node`), then unmounts and asserts `.current === null`. Without the cleanup half, `useStickyScroll` and search-scope consumers would hold dangling node pointers after navigation between Dashboard branches.
